Doc/library/signal.rst

```diff
@@ -503,3 +503,38 @@ be sent, and the handler raises an exception. ::
 
    signal.alarm(0)          # Disable the alarm
 
+Note on SIGPIPE
+---------------
+
+Do not set set :const:`SIGPIPE`'s disposition to :const:`SIGDFL`
+in order to avoid the :code:`BrokenPipeError: [Errno 32] Broken
+pipe` error message when you pipe your program into head(1).
+
+This will cause your program to exit in unexpected ways because Python
+will exit if a write to any socket which is the remote side has closed
+or dropped occurs.
+
+Instead wrap your main with a catch for IOError as such: ::
+
+    import signal
+    import os
+    import sys
+
+    def main():
+        for x in range(10000):
+            print("y")
+
+    if __name__ == '__main__':
```

Comment:
Two issues:
- this combines example code with test code: the flushes are there to force the issue, while the Exception handler is there solve the issue. A user might think they need the flushes.
- It is more typical, in code I've seen, to have nothing under the `if __name__` check other than a call to main and to have setup code happen in `main` or similar.  Note that this might not be standard python style.

Reply:
> this combines example code with test code: the flushes are there to force the issue, while the Exception handler is there solve the issue. A user might think they need the flushes.

The user does need the flushes, otherwise some stdout may be buffered until after the python program exits the try/except block.  You can demonstrate this by removing the flush and piping this program into `head -1`.

That said the flush on stderr is probably not needed and I can remove it.

I've spun another version moving the code into main() and removing the flush of stderr.  The flush of stdout must remain as far as I can tell.